users/views.py

```diff
@@ -164,4 +171,279 @@ def get(self, request, user_id, token):
+            }
+        )
+
+        return self.generate_jwt_response(user)
+    
+    def handle_github_oauth(self, access_token):
+        """
+        Authenticate using GitHub OAuth access token.
+        
+        1. Validates token with GitHub's API
+        2. Retrieves user profile and primary email
+        3. Creates/updates local user record
+        4. Issues JWT tokens
+        
+        Args:
+            access_token: Valid GitHub OAuth access token
+            
+        Returns:
+            Response: JSON containing JWT tokens and user data or error message
+            
+        Status Codes:
+            - 200 OK: Successful authentication
+            - 400 Bad Request: Invalid token or missing email
+        """
+        headers = {
+            "Authorization": f"Bearer {access_token}",
+            "Accept": "application/json"
+        }
+        
+        try:
+            # Get user info
+            user_response = requests.get(
+                "https://api.github.com/user", 
+                headers=headers
+            )
+            user_response.raise_for_status()
+            user_data = user_response.json()
+            
+            # Get email (might need separate request)
+            email = user_data.get("email")
+            if not email:
+                email_response = requests.get(
+                    "https://api.github.com/user/emails", 
+                    headers=headers
+                )
+                email_response.raise_for_status()
+                emails = email_response.json()
+                primary_email = next(
+                    (e for e in emails if e.get("primary")), 
+                    None
+                )
+                if primary_email:
+                    email = primary_email.get("email")
+
+            if not email:
+                return Response(
+                    {"error": "Email not provided by GitHub"},
+                    status=status.HTTP_400_BAD_REQUEST
+                )
+                
+            # Process name
+            username = user_data.get("login")
+            name_parts = user_data.get("name", "").split(" ", 1)
+            first_name = name_parts[0] if name_parts else ""
+            last_name = name_parts[1] if len(name_parts) > 1 else ""
+
+            user, _ = self.get_or_create_user(
+                email=email,
+                defaults={
+                    # "username": username,
+                    "first_name": first_name,
+                    "last_name": last_name,
+                    "password": "",
+                    "user_phone": "",
+                    "title": "",
+                    "role": get_default_role()
+                }
+            )
+
+            return self.generate_jwt_response(user)
+            
+        except requests.RequestException:
+            return Response(
+                {"error": "Invalid GitHub token"}, 
+                status=status.HTTP_400_BAD_REQUEST
+            )
```

Comment:
Add a timeout to both requests.get calls.
Log GitHub API errors for easier debugging.
Check for verified email status in the email selection logic.
Consider handling the case where "name" is empty or not in "first last" format.